[DONE] [CR] Field dressing corpses aka Butchery overhaul

nexusmrsep opened the pull request:
### **What does it do?**

This PR introduces **field dressing** and **full butchery**, **quartering** and **dissecting** of hunted game/corpses. 
Field dressing is a corpse preparation action, that may result in benefits like reduced weight for transportation and reduced rot, as intestines and organs are removed from the body and are not longer a threat to the quality of meat.

Idea was introduced in #24145 as a result of a long discussion.

### **Racionale:**
[Field dressing](https://en.wikipedia.org/wiki/Field_dressing_(hunting))
Another source [HERE](https://books.google.pl/books?id=kh8NtAVg_k4C&pg=PA70&lpg=PA70&dq=butchering+hanging+rack&source=bl&ots=HCVKEFSuVc&sig=uQULnuS7mdRuZPQAGxaUNyRBxko&hl=pl&sa=X&ved=2ahUKEwj20vWsksTcAhXJBiwKHXkmBfEQ6AEwDHoECAcQAQ).

### **Features**
**[heavily edited to include current approach]**
1. **Field dressing**, **quick** and **full butchery** and **quartering** (+ **dissecting** ) are all sub-types of butchery action accessible form the butcher menu.
3. Field dressing affects butchery yields removing internal organs like stomach/offal from the list of possible results - and yields internal organs (offal for example).
4. Field dressing can be performed with varying results, from fully successful to a bloody mess (with gibs flying overhead). Failed results might destroy or fairly damage the corpse, and significantly affect butcher results.
5. Successful field dressing reduces initial rot of harvested meat, and reduces rot rate of the corpse itself.
6. Field dressed corpse is a (carcass), quartered field dressed corpse is a (quartered carcass).
7. You can **quarter** field dressed corpse into four parts - applies to volume and mass, equals easier transportation. Might not help for huge creatures, as they are HUGE! Tiny creatures can't be quartered. Quartering ruins skin, so a confirm popup prevents unwanted destruction of valuable resource if not desired by the player.
8. **Quick butchery** yields up to 1/4 of max corpse yields in flesh - its just a quick harvest before moving on. It destroys skin and anything else - what do you care if you just need few pieces of meat?
9. **Full butchery** can be performed in a workshop consisting of a butchers rack (new construction - few long sticks stitched together with some ropes or wires to hang the carcass in the air), a nearby table, and requires extra tools, mainly sawing tools and cutting tools (expl.: some low-grade butchery tools are not even cutting tools).
10. Full butchery takes significantly more time, but yields full potential of a corpse.
11. Meat yields were **significantly** increased, influencing some other yields too. [may need balancing] See comments.
12. You don't have to field dress a corpse to perform full butchery, but it will take longer, and you might have trouble if you want to drag your overgrown bear to the butchers workshop.
13. Morale now affects the possibility of butchery. Depressed person would not engage in bloody work.
14. **CBM removal** from corpses was moved to **DISSECT** option only. **EDIT:** see comments below for more features of this particular option.
15. You **have to have** a fine cutting tool to remove CBM's while butchering.
16. You can ~~no longer~~ butcher human corpses [edit: at a morale penalty] unless you are a psycho, cannibal, or sapiovore.
17. Few different variations of result messages for better flavour.
18. Gibs and gore. Plenty of.
19. Mobile workshop for nomads (see comments): folding metal butchering rack, folding metal tourist table, also kitchen unit in car works as a table equivalent.

### **Implementation:**
- mainly invasive reworks in `butcher` activity handlers

### **TODO**:
- [x] core concept foundations
- [x] finishing all the expected effects, conditions, results,
- [x] debugging,
- [x] heavy testing,
- [x] balancing,
- [x] cleanup

nexusmrsep commented:
> If I get this right, the field dressing in game terms means the same butcher, just without intestines?

Field dressing is an impartial butcher. Let my quote Kevin from #24145 here:

> field dressing, discards less valuable parts, makes the corpse more compact for carrying, and delays spoilage. My understanding is it's fairly rare to butcher kills immediately when hunting so something like this is very typical.

So the target benefits are (not yet fully coded as this is Work-In-Progress):
- reduced decay of the corpse,
- reduced weight,
- reduced volume.

By field dressing you not only preserve the corpse that without it would quickly rot or became otherwise inedible, but have an easier job in transporting whole body to your hideout, and you may wait before you finally butcher it for meat. Price paid is removal of intestines and some bones (imagine partial removal of game's legs or head), so no stomach, no offal, but still all the meat, skin/fur, etc. Unless you mess it up and hack the body and spill the guts all over it.

In that way, yes, it is a method of butchering.

nexusmrsep commented:
> Honestly I haven't heard about anyone moving corpses somewhere. It is much more practical to butcher the corpse right where you killed it.

You had no other option before so its not that you had any other choice on that, or could observe or heard about a different behavior. But the point here is not to introduce if just for the sake of it, but to give such incentive to the player, that he might consider this as an option. Player might totally ignore it, if he chooses to, or simply decide that butchering a bear and getting just just a few chunks of meat is better then the tedium of field dressing, getting it on a travois, transporting to base, unloading, sticking it in a basement for reduced temperature rot, etc. just to be able to store raw meat for a week or two.

Also it's a [CR] becouse I wanted opinions and ideas how should it be balanced in comparison to simple butchering. Perhaps it should be easier, or have less depth that i already placed in the code, etc. Decay scaling can be discussed etc. There are many balancing points of such option.

But if there is no room for it in the vanilla tell me now before I invest another week in it.

sfsworms commented:
I frankly don't really see a reason to use it in game. IRL, you don't butcher the kill directly because you don't have the equipment needed with you (and it's easier to drag one carcass than to fill your backpacks with chunk of sanginolent meat). 

One way to replicate that would be to significantly increase the yield of meat if you butcher a corpse at a butcher shop, or some other bigger equipment. After all, right now, the amount of meat you get from butchering is very, very low. I just tested on some cougar I killed, and a 80 kg carcass yielded ~4 kg of meat and ~4 kg of bone, and I have decent survival (4) and a good knife. 

Quickly looking online, you can expect around 1/6 of live wieght in meat for a moose, 1/2 for a pig, and 1/3 for deet (that's a very quick look, if you want sources, I'll take the time to compare several websites). 

So that could be a balancing: you can either quickly harvest some meat with your knife, which should be enough if you're on the moove, or lack way to store the meat, or properly dress the carcass, bring it back to your base, and properly butcher it for a longer job, but a vast yield of meat. That second option would be worth it if you have the infrastructure to process and store a lot of meat, or if you have followers tof eed.

mlangsdorf commented:
Faction camp followers require a lot of food. Having the option to field dress a moose or bear (I have never successfully hunted deer in CDDA, though next time I'll try with a rifle instead of a bow) and then drop it off at the faction base kitchen for further processing would be great.

sfsworms commented:
I had quite a lot of success by chasing deer against a river bend and shooting tem with a bow from short range.

nexusmrsep commented:
> These are removed, but it seems that if you don't mind the weight they aren't necessarily ruined, so you can also just store them separate from the rest of the corpse. Ideally the field dressing operation will produce the byproducts and then let the player decide what to do with them, subject to the usual runiation chances.

That's the plan.

I would also remove the possibility of skinning a corpse on a non-field-dressed game and pass it to full-butchering field-dressed corpses. If butchering before field dressing is considered a quick way to obtain few pieces of meat, then you don't care about the skin/fur.

BTW, what kind of tools would full butchering need to be well performed? Should it be just high-quality butchering tools, or something more special? I'm thinking of a butchering rack that you tie the corpse to. It might be a possible passive requirement. And perhaps a table?

Also - should field dressing be an option available from the butchering action, or an action invoked based on item quality (butchering) as it was initially planned in this PR. There would be no problem for me to do it both ways, but I'm not sure which would be more convenient.

kevingranade commented:
> If butchering before field dressing is considered a quick way to obtain few pieces of meat, then you don't care about the skin/fur.

I agree

> BTW, what kind of tools would full butchering need to be well performed?

Some resources: https://honest-food.net/tools-for-home-butchering/
tl;dr a wide variety of cutting/sawing tools, table space, and hanging gear

> should field dressing be an option available from the butchering action

Meant to mention that earlier, I'd prefer to keep both associated with the butcher menu.

DracoGriffin commented:
>I foresee all kind of complaints like "Ooh, I like the previous simple and convenient behavior "stand on corpse -> B -> done!".

They'll just have to adapt like they have with knife spears, health regeneration (bandages and such) and autodoc changes.

If anything, it could be useful to get ahead of it and make announcements of the changes, and offer advice on the adjustment rather than a blind update where all of a sudden they have to relearn the whole butchering mechanics (which as a player, I can totally understand and relate).

sfsworms commented:
Well, as I understand it, the full plan isn't to remove the 'B' -> butcher mechanic, but to add another one, which is more complicated (Dress the corpse, build a butcher stand or something, bring the corpse to the stand, butcher it) that yields more meat. I don't really foresee complains.

While I'm at it, can I also propose that some way to quarter large carcasses be provided? That's also frequently done. It could even be integrated in the field dressing, where carcasses over a certain weight are automatically quartered in addition to being dressed.

Night-Pryanik commented:
> Well, as I understand it, the full plan isn't to remove the 'B' -> butcher mechanic, but to add another one, which is more complicated (Dress the corpse, build a butcher stand or something, bring the corpse to the stand, butcher it) that yields more meat. I don't really foresee complains.

Look at the latest comments in #24243. The added functional increased keypresses, and there were complaints on that.

nexusmrsep commented:
I'll expand the menu for butchering, so each corpse will show as two entries: 'Butcher foo', 'Field dress foo', and a "Field dress every corpse" option. This might be convenient enough, but this also means that I have to force it to popup even on a single corpse.

nexusmrsep commented:
I'm putting meat yield rationale under discussion. Tell me what you think:
I started from corpse's in-game weight (40 kg dog is a monster!):

            case MS_TINY:   ret =   1000_gram;  break;
            case MS_SMALL:  ret =  40750_gram;  break;
            case MS_MEDIUM: ret =  81500_gram;  break;
            case MS_LARGE:  ret = 120000_gram;  break;
            case MS_HUGE:   ret = 200000_gram;  break;

then knowing that one meat chunk = 200_gram, and using [THIS ](http://www.butcher-packer.com/index.php?main_page=document_general_info&products_id=331) source to determine that meat yield is roughly equal to 30-35% full body weight of animal before field dressing I propose following yields:

     PRESENT (MAX) --> TARGET BUTCHERING (MAX) --> TARGET 'WORKSHOP' BUTCHERING (of field dressed corpse)

        case MS_TINY:
            return 1;  200_gram  --> 1; 200_gram --> 2; 400_gram (40% body mass) //unsuitable for field dressing?
        case MS_SMALL:
            return 2;  400_gram --> 16; 3200_gram --> 64; 12800_gram (31% body mass)
        case MS_MEDIUM:
            return 4;  3200_gram --> 32; 6400 --> 128; 25600_gram (31% body mass)
        case MS_LARGE:
            return 8;  1600_gram --> 48; 9600_gram --> 192; 38400_gram (32 % body mass)
        case MS_HUGE:
            return 16; 3200_gram --> 80; 16000_gram --> 320; 64000_gram (32 % body mass)

So as you can see the proposed quick butcher to workshop butcher ratio is **1:4**.
Those numbers seems big, but that's simply math results.

nexusmrsep commented:
> While I'm at it, can I also propose that some way to quarter large carcasses be provided? That's also frequently done. It could even be integrated in the field dressing, where carcasses over a certain weight are automatically quartered in addition to being dressed.

I can apply it by adding a `QUARTERED` flag and multiply quartered body times factor of quartering.
If this factor = 4 (like the word 'quartering' states) for all corpses (excluding tiny) then each of the 4 parts will have weight/4, volume/4. This however does not scale so `MS_HUGE` corpses will be:
- weight: 200000_gram x 0.75 (field dressing) / 4 (quartering) = 37,5 kg
- volume: 875000_ml / 4 = 218 liters. (volume is not affected by field dressing as far as I know)

Is this acceptable as far as those numbers go?

mlangsdorf commented:
A single chukkar or pheasant is a meal for a single person, and they're MS_TINY. The yields on the low end look about right, and on the high end, a single moose is generally enough food for one person for a year if it gets fully butchered.
MS_TINY critters should be eligible for field dressing: that represents a quick cleaning to get the breast meat while removing the intestines. People who are frugal could then complete the butcher to get the rest of the meat. Speaking from personal experience here from hunting pheasants: most hunting guides will field dress the birds and give you the breast meat, but you can roughly double your take by butchering the birds yourself and recovering the leg meat (and intestines/gizzard/heart, if you're into that sort of thing).

Butchering equipment should scale with critter size. You can completely butcher a bunch of pheasants with a hatchet, a pair of shears, and a bucket of water, while fully butchering a deer or bear should take additional tools.

nexusmrsep commented:
Another question: Should CBM recovery be left untouched or moved to full workshop butchery only?

Robik81 commented:
> I'll expand the menu for butchering, so each corpse will show as two entries: 'Butcher foo', 'Field dress foo', and a "Field dress every corpse" option. This might be convenient enough, but this also means that I have to force it to popup even on a single corpse.

I don't think we need to of force popup on every corpse, should be only on specific corpse types that have the field dress option. Zombies and small animals like rabbits don't need such options.

sfsworms commented:
Why not? What if you want a lot of tainted meat for mutagen?

Re: CBM, I think it makes sense to keep them for field butchering. You don't need to carefully save all the meat, you're just ripping the CBM from the body

Robik81 commented:
> Why not? What if you want a lot of tainted meat for mutagen?

You can just kill more zombies, its not like they are in short supply.

nexusmrsep commented:
Exactly, original idea was to trim it to WILDLIFE plus humans for canibals, but due to the agreement on meat yields being higher on "workshop" butcher and including how broad the spectrum is, it can't stay.

kevingranade commented:
CBM recovery would be either:
Part of full butchery.
Dedicated action that ruins the corpse for other uses.

Night-Pryanik commented:
> Dedicated action that ruins the corpse for other uses.

It may be good for balance reasons, but why removing, say, Cerebral Booster (obviously located in a head), should ruin the whole corpse?

sfsworms commented:
Does it make sense as part of the butchery? I mean even the tools used in butchery aren't great to remove cbms, you'd expect to need fine cutting for that.

Night-Pryanik commented:
> Does it make sense as part of the butchery? I mean even the tools used in butchery aren't great to remove cbms, you'd expect to need fine cutting for that.

I suppose full butchery will require tools with fine cutting.

kevingranade commented:
> why removing, say, Cerebral Booster (obviously located in a head), should ruin the whole corpse?

Mostly because it definitely ruins some harvestables (in this example part of the brain), and it's not IMO worthwhile to track how much and have a system for tracking which CBMs have been removed.

If you have the incredibly niche desire to harvest both meat and CBMs from the same body, do it all at once.

Night-Pryanik commented:
> If you have the incredibly niche desire to harvest both meat and CBMs from the same body, do it all at once.

Fully agree, that's why I prefer your first variant (`Part of full butchery.`) over the second one.

Night-Pryanik commented on data/json/construction.json:
```diff
@@ -2279,5 +2279,25 @@
+        ],
+        "components" : [
+            [ [ "stick_long", 6 ], [ "2x4", 6 ] ],
+            [ ["string_36", 8 ], [ "rope_makeshift_6", 2 ], [ "rope_6", 2 ], [ "wire", 8 ], [ "vine_30", 1 ] ],
```
Ropes could be replaced with `rope_natural` or `rope_natural_short` requirements.

Night-Pryanik commented on data/json/furniture.json:
```diff
@@ -2987,5 +2987,33 @@
+        "examine_action": "reload_furniture",
+        "deconstruct": {
+            "items": [
+            { "item": "stic_long", "count": 6 }
```
stick_long

Night-Pryanik commented on src/activity_handlers.cpp:
```diff
@@ -682,11 +736,55 @@ void activity_handlers::butcher_finish( player_activity *act, player *p )
+            }
+            if( ( !p->has_quality( quality_id( "CUT" ) ) || !p->has_quality( quality_id( "CUT_FINE" ) ) ) &&
+                ( !p->has_quality( quality_id( "SAW_W" ) ) || !p->has_quality( quality_id( "SAW_M" ) ) ) ) {
+                p->add_msg_if_player( m_info, _( "You need cools that can cut and saw to perform a full butchery." ) );
```
tools

nexusmrsep commented on data/json/construction.json:
```diff
@@ -2293,7 +2293,7 @@
         "components" : [
             [ [ "stick_long", 6 ], [ "2x4", 6 ] ],
-            [ ["string_36", 8 ], [ "rope_makeshift_6", 2 ], [ "rope_6", 2 ], [ "wire", 8 ], [ "vine_30", 1 ] ],
+            [ [ "rope_natural", 1 ], [ "wire", 8 ] ],
```
This:

        "using": [ [ "rope_natural", 1 ] ],

while being viable for `recipe_others.json` will not work for `construction.json`, so I'm reverting it to original, at least for now.

nexusmrsep commented:
**Status update:**
- updated PR name, as it's gone wider and deeper then initial anticipated (I see a trend here),
- updated/edited initial description to sum-up important changes of the new approach.

**As for the progress:**
- main concepts are coded, menus are working as intended
- it compiles, first round of tests finished, some post-test reworks were applied, but there is a lot to do, as not everything works as intended. I will get there eventualy.

Xhuis commented:
Are there vehicle parts allowing for full butchery? Nerfing the ability to get meat from corpses for a nomadic lifestyle is severely hampering to their effectiveness.

Xhuis commented on src/activity_handlers.cpp:
```diff
@@ -242,7 +255,47 @@ void set_up_butchery( player_activity &act, player &u )
+    }
+    // applies to all butchery actions
+    if( corpse->in_species( HUMAN ) && ( !u.has_trait_flag( "CANNIBAL" ) || !u.has_trait_flag( "PSYCHOPATH" ) || !u.has_trait_flag( "SAPIOVORE" ) ) ) {
+        add_msg( m_info, _( "Why would you do this to mortal remains of a fellow human?" ) );
```
Personal thought: This should still be allowable, but give a morale penalty if you're still bent on preventing it. Historically this kind of thing has been skewed towards "allow the player to do detrimental things if they really want to", like how you can consume human meat (even if it gives a huge morale penalty) and the like.

nexusmrsep commented:
@Xhuis thanks for the ideas. I will look into them along the way. I believe kitchen unit in a vehicle might be a viable option though I can't imagine how you could fit a bear in a car and butcher it easily in a confined space. Maybe I'll add a portable metal folding  butcher's rack, that can be unfolded and set on a ground as an advanced option for nomads.

LadyCatnip commented:
> It may be good for balance reasons, but why removing, say, Cerebral Booster (obviously located in a head), should ruin the whole corpse?

I imagine that it has to do with the fact that the player doesn't know where all the CBM's are, so they just go through the entire body to find them, while also ruining the meat.

Night-Pryanik commented:
> I imagine that it has to do with the fact that the player doesn't know where all the CBM's are, so they just go through the entire body to find them, while also ruining the meat.

Well, that's a good point, I buy it.

sfsworms commented:
> Are there vehicle parts allowing for full butchery? Nerfing the ability to get meat from corpses for a nomadic lifestyle is severely hampering to their effectiveness.

I don't think that is a good idea. There has been a lot of discussion about how to make sedentary bases viable, even if only temporarily. Forcing nomads to set up a base camp for hunting make perfect sense. You can still get some meat on the go via field butchering.

The one issue is for CBMs, hauling zombies corpses to the butcher stand is going to be a pain.

Mecares commented:
> I don't think that is a good idea. There has been a lot of discussion about how to make sedentary bases viable, even if only temporarily. Forcing nomads to set up a base camp for hunting make perfect sense. You can still get some meat on the go via field butchering.

Unrealistic limitations to vehicles are not a good way to make bases more appealing, if something can be done plausible and realistic in vehicles, then  it should not be prevented.

cainiaowu commented on src/game.cpp:
```diff
@@ -10147,13 +10179,37 @@ void game::butcher()
         break;
     case BUTCHER_CORPSE:
         {
+            uimenu smenu;
```
Adding additional menu might cause player complaining about keypress, like the last time when hoe getting a digging function.

nexusmrsep replied:
Believe me, this is the minimal approach that could be done. I even reworked the initial commits to reduce the clutter. If you have a better idea how to make this choice without asking a player tell me.

cainiaowu replied:
My idea is that the game should remember the butcher mode and player can switching between them.
So normal usage will be "Select mode -> butcher however many corpses you like -> Select another mode". 
Just like the gun firing mode.

cainiaowu commented:
What will happen to Auto Pulping mode? It has auto butcher function integrated in it.

nexusmrsep commented:
Auto Pulping should stay auto. I will test it if it stays this way and if it doesnt I will rebuild it in that it stays conceptualy the same. This is of course quick butchery.

nexusmrsep commented:
@sfsworms and @Mecares - by adding foldable rack that is not a part of vehicle you can achieve both - "setting camp" by unfolding and placing rack outside vehicle and having vehicle's kitchen unit instead of a table. Balanced, and reasonable.

Night-Pryanik commented on src/activity_handlers.cpp:
```diff
@@ -242,7 +255,47 @@ void set_up_butchery( player_activity &act, player &u )
 
+    bool has_table_nearby = false;
+    for( const tripoint &pt : g->m.points_in_radius( u.pos(), 1 ) ) {
+        if( g->m.furn( pt ) == furn_id( "f_table" ) ) {
```
Don't hardcode this to table only. Add a flag like `TABLE` to the flat surface furniture and check for it here instead.

Night-Pryanik commented on src/activity_handlers.cpp:
```diff
@@ -242,7 +255,47 @@ void set_up_butchery( player_activity &act, player &u )
+    }
+    // workshop butchery (full) prequisites
+    if( action == BUTCHER_FULL ) {
+            if( g->m.furn( u.pos() ) != furn_id( "f_butcher_rack" ) ){
```
The same here. In case you're planning to add some portable variant of butchering rack, add a flag to every variant and check for it here.

nexusmrsep replied:
In light of current discussion its the best way to do this.

nexusmrsep commented:
@cainiaowu while this sounds like a fine idea, problem lies in where to put this option (variable) and how to change it in play. On one hand you don't want to be prompted every time to confirm that you still want to use default action, and on the other hand pushing this to general Options will be counter intuitive (you will forget that there is more then once chosen default) and a terrible hassle. You want to field dress a corpse? Goto options, tab nr 5, option nr 123, change, save, now you can can field dress. Then you want to do something else and go over the whole process again. See where this is going? Terrible UI. Terrible design. I'm not against making it less clunky but I have seriuos doubts about way of achieving it. Also multibutcher still exists in this implementation so you can butcher all you can with quick butchery.

Inglonias commented:
I have mixed feelings about this.

> I foresee all kind of complaints like "Ooh, I like the previous simple and convenient behavior "stand on corpse -> B -> done!".

I feel exactly this way. Legitimately. That said, I also see the benefit of this approach. If you want to do this, do it right. A few things need to happen here.

First, all butchers shops need to have a butcher's table in the back for people to deconstruct. Second, you need to have a vehicle version of the butcher's table that can be built. Finally, you should still be able to get some meat off of corpses just by field dressing it. Penalize it, keep it the same, I don't care. Just allow for something.

cainiaowu commented:
IIRC you change your guns fire mode via single key press "f".

nexusmrsep commented:
@Inglonias Buchering rack in butcher shop - great idea. Vehicle butcher table - unneccesary, kichen unit will be enough or maybe other existing parts with flat surfaces. Field dressing yields (already accounted for): yes for stomach, offal and some bones, but no for meat. Field dressing has nothing to do with harvesting meat. See sources.

@cainiaowu I have a mixed feeling about populating and locking another global key for mode switching just for butchery. Perhaps 'f' could be a menu of modes used in different things (like the hoe problem) but this would be outside the scope of this PR and would need to be done separately.

sfsworms commented:
If you keep the field butchering keyed to 'b' per default, people that want to experience the statu quo will just have to double-tap b instead of having multiuple keypresses. The added tedium is minimal.

nexusmrsep commented:
Thank you all for constructive discussion and reviews. Let's sum up new features/reworks:
- you can butcher human corpse even when not being psychopath/cannibal/sapiovore, at a big morale penalty and you will be prompted to confirm this action.
- added metal butcher rack, it's a furniture that can be deconstructed to an item for transportation, and can be mounted (or deployed) on a floor elsewhere, it is a standard equipment of butcher's shop
- BUTCHER_EQ flag for both wooden and metal butchering rack
- FLAT_FURN flag for ... guess what ... hard flat surfaces, now added to table, desk, counter, pool table. Thinking about adding it to other flat but otherwise strange to use options like: washing machine, dryer, coffin (sic!) - this is used as a condition for full butchery.
- vehicle's kitchen unit in now equal to a furniture with FLAT_FURN flag, as a requirement for full butchery.

After all this it is still in testing phase. As always I wish I had more time.

Night-Pryanik commented on data/json/construction.json:
```diff
@@ -2297,5 +2297,20 @@
+            "components" : [
+                [ [ "metal_bucher_rack", 1 ] ]
+            ],
+            "pre_note" : "It's legs are foldable and hold in place by screws.",
```
Its.

Dansiman replied:
held

Rivet-the-Zombie commented:
>You can no longer butcher human corpses unless you are a psycho, cannibal, or sapiovore.

Why not leave players the option, but cause a morale penalty for doing so unless you meet the stated criteria?

ymber commented on data/json/furniture.json:
```diff
@@ -2988,6 +2988,57 @@
+        "deployed_item": "metal_butcher_rack",
+        "examine_action": "deployed_furniture",
+        "flags": ["PLACE_ITEM", "TRANSPARENT", "MOUNTABLE", "ALLOW_FIELD_EFFECT", "BUTCHER_EQ"],
+        "deconstruct": {
```
Why use both the construction system and furniture deployment for this? Would be simpler to pick one.

nexusmrsep replied:
I agree, but which one would be better? I couldn't decide so I left both for testing, before removing one of them. So far deployment method is slightly more annoying becouse it pops up a query if you want to disassemble the item every time you want to check the tile for butchery results/yields.On the other side I believe that bloating construction menu with simple tasks is not a good thing either.

ymber replied:
I'd go with furniture deployment. It's simpler and more intuitive than using the construction menu and it's a very minor inconvenience when it queries for disassembly.

DracoGriffin replied:
I actually like the idea of both; one for early butcherers and a more advanced "travel" type. However the weight/volume might need to be adjusted.

nexusmrsep commented:
Question: is there a need for a folding deployable turist table with FLAT_FURN that can be taken as a baggage in a vehicle and serve as a mobile table for full butchering along with the metal folding butchery rack?

nexusmrsep commented:
I believe it is done at this point.

Since this is a massive PR please test this to your heart content before merging and point out any discrepancies, etc. Have fun doing so. I might also do some extra rounds of testing myself. 

Be aware that some yields might be not balanced perfectly, but to my belief they seem fine for now.

DracoGriffin commented on data/json/furniture.json:
```diff
@@ -2988,6 +2988,84 @@
+        "type" : "furniture",
+        "id" : "f_butcher_rack",
+        "name": "butchering rack",
+        "description": "Butchering rack designed to hang the carcass in the air.",
```
Suggestion changing `hang the carcass` to `hang a carcass` or `hang carcasses`.

DracoGriffin commented on data/json/furniture.json:
```diff
@@ -2988,6 +2988,84 @@
+        "type" : "furniture",
+        "id" : "f_metal_butcher_rack",
+        "name": "metal butchering rack",
+        "description": "Metal butchering rack designed to hang the carcass in the air.  It can be deconstructed and folded for easy transportation.",
```
Suggestion changing `hang the carcass` to `hang a carcass` or `hang carcasses`.

DracoGriffin commented on data/json/furniture.json:
```diff
@@ -2988,6 +2988,84 @@
+        "type" : "furniture",
+        "id" : "f_tourist_table",
+        "name": "tourist table",
+        "description": "Small metal folding table, ideal for off-road trips to the wild.",
```
Not sure if `into the wild` sounds better than `to the wild`, but a minor suggestion nonetheless.

DracoGriffin commented on data/json/items/tool/deployable.json:
```diff
@@ -13,5 +13,35 @@
+    "id": "metal_butcher_rack",
+    "type": "TOOL",
+    "name": "metal butchering rack",
+    "description": "Metal butchering rack designed to hang the carcass in the air.  It is folded for easy transportation and can be deployed as a furniture.",
```
Suggestion changing `hang the carcass` to `hang a carcass` or `hang carcasses`.

DracoGriffin commented on data/json/morale_types.json:
```diff
@@ -279,6 +279,11 @@
+  {
+    "id": "morale_butcher",
+    "type": "morale_type",
+    "text": "Memories of Butchering Human Corpse"
```
Suggest `[Appalled/Anguished] for Butchering [Fellow] Human` --- [ ] are optional/options, "fellow" might not always be the case with cyborgs/supermutants/mycus.

DracoGriffin commented on doc/JSON_FLAGS.md:
````diff
@@ -57,6 +57,7 @@ List of known flags, used in both terrain.json and furniture.json
 - ```BARRICADABLE_WINDOW``` Window that can be barricaded.
 - ```BARRICADABLE_WINDOW_CURTAINS```
 - ```BASHABLE``` Players + Monsters can bash this.
+- ```BUTCHER_EQ``` Butcher's equipment - required for full butchery of corpses
````
Minor correction: missing period at end

DracoGriffin commented on src/activity_handlers.cpp:
```diff
@@ -242,7 +256,83 @@ void set_up_butchery( player_activity &act, player &u )
+            }
+            if( ( !u.has_quality( quality_id( "CUT" ) ) || !u.has_quality( quality_id( "CUT_FINE" ) ) ) &&
+                ( !u.has_quality( quality_id( "SAW_W" ) ) || !u.has_quality( quality_id( "SAW_M" ) ) ) ) {
+                u.add_msg_if_player( m_info, _( "You need tools that can cut and saw to perform a full butchery." ) );
```
A better catch-all statement since almost all fine cutting tools have cutting:
`You needs tool with fine cutting and sawing to perform a full butchery.`

Should help reduce the amount of confusion when players are trying to extract CBMs.

DracoGriffin commented on src/activity_handlers.cpp:
```diff
@@ -242,7 +256,83 @@ void set_up_butchery( player_activity &act, player &u )
+    bool is_human = corpse == nullptr || corpse->id == mtype_id::NULL_ID() || ( corpse->in_species( HUMAN ) && !corpse->in_species( ZOMBIE ) );
+    if( is_human && !( u.has_trait_flag( "CANNIBAL" ) || u.has_trait_flag( "PSYCHOPATH" ) || u.has_trait_flag( "SAPIOVORE" ) ) ) {
+
+        if( query_yn( "Would you do this to mortal remains of a fellow human?" ) ) {
```
Suggest: `Would you dare desecrate the remains of a fellow human?`

DracoGriffin commented on src/activity_handlers.cpp:
```diff
@@ -486,47 +648,59 @@ void butchery_drops_hardcoded( const mtype *corpse, player *p, const time_point
+                ( corpse->has_flag( MF_CBM_CIV ) || corpse->has_flag( MF_CBM_SCI ) ||
+                corpse->has_flag( MF_CBM_TECH ) || corpse->has_flag( MF_CBM_SUBS ) ||
+                corpse->has_flag( MF_CBM_OP ) || corpse->has_flag( MF_CBM_POWER ) ) ) {
+        p->add_msg_if_player( m_bad, _( "You found bionics implanted in this corpse, but without fine cutting tools you ruin them completly." ) );
```
Correction: `completely`

DracoGriffin commented on src/activity_handlers.cpp:
```diff
@@ -611,11 +879,32 @@ void activity_handlers::butcher_finish( player_activity *act, player *p )
+    const field_id type_gib = corpse->gibType();
+
+    if( action == QUARTER ) {
+        if( query_yn( _( "Quarter corpse of %s to reduce it's weight and volume?  This will ruin it's skin." ), corpse->nname().c_str() ) ) {
```
Correction: `its`
However, I'd suggest: `Quarter corpse of %s to reduce the weight and volume?  This will ruin the hide or pelt.`

DracoGriffin commented on src/activity_handlers.cpp:
```diff
@@ -631,30 +920,131 @@ void activity_handlers::butcher_finish( player_activity *act, player *p )
+                p->add_msg_if_player(m_warning, _( "You wanted to cut the corpse, but instead you hacked the meat, spilled the guts all over it, and made a bloody mess." ) );
+                break;
+            case 3:
+                p->add_msg_if_player(m_warning, _( "You made so many mistakes during the process that you doubt even vultures will be intrested in what's left of it." ) );
```
Correction: `interested`

DracoGriffin commented on src/activity_handlers.cpp:
```diff
@@ -631,30 +920,131 @@ void activity_handlers::butcher_finish( player_activity *act, player *p )
+    case QUARTER:
+        break;
+    case BUTCHER:
+        p->add_msg_if_player( m_good, _("After quick attempt to harvest flesh from %s you leave the rest for the vultures."), corpse_item.tname().c_str() );
```
Suggestion: `After quickly harvesting any usable meat from %s, you leave the unsalvageable mess for scavengers and vultures.`

nexusmrsep replied:
Can't use word 'meat' because of tryffids etc. Flesh is better. Also "After quickly harvesting" suggests that it was a success, but its not guaranteed here. But I've got a different idea how to wite this line, so check it out.

DracoGriffin commented on src/activity_handlers.cpp:
```diff
@@ -631,30 +920,131 @@ void activity_handlers::butcher_finish( player_activity *act, player *p )
+        if ( roll_butchery() < 0 ) { // partial failure
+            switch( rng( 1, 3 ) ) {
+                case 1:
+                    p->add_msg_if_player(m_warning, _( "You unskilfully hack the corpse and chop off some excess body parts.  You think this was not a job well done." ) );
```
Correction: `unskillfully`
Suggestion: `You unskillfully hack up the corpse and chop off some excess body parts.  You're left wondering how you did so poorly.`

DracoGriffin commented on src/activity_handlers.cpp:
```diff
@@ -631,30 +920,131 @@ void activity_handlers::butcher_finish( player_activity *act, player *p )
 
+        }
+        if( corpse->size > MS_TINY &&
+            query_yn( _( "Quarter %s to reduce it's weight and volume?  This will ruin it's skin." ), corpse_item.tname().c_str() ) ) {
```
Correction: `its`
Suggestion: `Quarter corpse of %s to reduce the weight and volume?  This will ruin the hide or pelt.`

nexusmrsep commented:
@DracoGriffin  - thanks for reviewing.
I will try to answer in parts:

> Quick Butchery should have Field Dress baked in

Absolutely not! Those are conceptually different actions. If you are in a hurry and just want some meat (or a pulp corpse equivalent) you just do a quick butcher & aim for major muscles. If you are in a hurry but still want some meat **and** some offal than you need to **first**: field dress the corpse, **then** quick butcher it. Those actions do not interfere with each other if done in this order, but if you do it the other way quick butcher will ruin the corpse. Field dressing do not yield CBMs as you think it does.

> I suggest re-ordering the butcher menu to:
Field dress corpse
Quarter carcass
Quick butchery
Full butchery

I will not do that - the reason behind is in the discussion above - adding a new menu is already non-convenient so there was a request to keep the most common option first and linked to "Q" so that Q->Q would be convenient option for quick butcher.

> add a subprompt like Martial Arts menu does by pressing F1 to get some additional info 

Will try to do that, as soon as I learn how it works. I'm not familiar to the methods this is achieved yet.

>Full butchery needs a chance to yield CBMs as well, it seems a bit strange that it doesn't unless I overlooked the rationale. Maybe you lose a bit of meat, but it seems like CBMs would be the most valuable item yielded from a corpse.

In fact **ONLY** full butchery can yield CBMs, no other method does.

> Using "butcher all" option needs to default to performing "full butchery", and if not on top of butchering rack / nearby, resort to using "quick butchery"

That would be in my opinion very hard to achieve, but not impossible. Multi-butcher option is mainly for dealing with zombies as an alternative to pulping. Also I had to remember that auto-butchering is a thing and it has to work.

I might try to work on it if you want, but I guarantee no success and it will put this PR on a hold for some weeks.

> Another issue is any monster that has a harvest.json defined, may be giving incorrect output items (and sometimes doesn't seem to work at all).

Disagree. Don't expect to get every item from the list unless you full butcher untouched corpse. It goes through every item on the list and checks if its valid for the type of butchery you are performing. If it's not - it will be ignored.  To my belief it works fine unless proven otherwise - take note that I deliberately hidden feedback like "Failed to harvest: stomach" when full butchering previously field dressed corpse, because it would be misleading (not present is not a failure). Only perceivable problems I can think of is that it might have problems dealing with uncommon items. But I still have to push some broken cyborgs through it and check the results. There might be also rounding problems like 5 / 4 = 1 (so you lose 1 of something in the process), but that's a minor problem.

nexusmrsep commented:
@DracoGriffin  after more tests using shocker brutes and broken cyborgs as examples, as you suggested, I confirmed the problem with CBM handling and fixed it. It's now correctly yielding CBM-s no matter if they are hardcoded or not. I also added hints for other butchery methods, to tell the player that he encountered traces of CBMs, so he can try full butchery next time. Non-standard items should be properly handled by the butchery, but in the future adding some weird ones might need some code reworks. One problem that I see is that broken cyborg has an awfuly small chance of yielding CBMs in my opinion, so it might need some tweaking after this PR gets in.

nexusmrsep commented:
@DracoGriffin  following your guidelines I have added menu descriptions outlining the expected results of each butchering technique. That was a good call, and thanks for the idea. Players will be much more aware of what to expect now.

Night-Pryanik commented:
The PR is becoming huge. It's approaching one thousand lines. It will probably be hard to review.

nexusmrsep commented:
@DracoGriffin  - it seems that I misjudged the difficulty of allowing multi-butcher action to be a full-butchery on a butchering rack. It as actually very easy, instead of very hard. Feature added, enjoy.

@Night-Pryanik  - I'm aware of it. Unfortunately, this couldn't be chopped to parts like you did with the autodoc, because there are a plethora of interdependence in this... sorry. But luckily this is the end, it's finished, can be tested, and only things I add as of now are changes requested by reviewers.

tyrael93 commented:
I have to say I personally like the idea of having to have more steps to extract the CBMs from corpses, but wouldn't it be really tedious to deploy a butchering rack every time you find a shocker zombie? This looks like one of those changes that look better on paper, since nobody really likes carrying around a bulky item for just in case you'll end up finding a cbm-bearing corpse, just for a chance of getting another power storage CBM.

sfsworms commented:
It's also not clear how a butcher rack helps harvest CBMs. Why do you need to hang your zombie for that? If anything, logical requirements would be some kind of fine cutting tool and maybe a table, so you can carefully remove the CBMs.

nexusmrsep commented:
Sometimes a picture is worth 1000 words. Can you imagine **not** hanging this for full butchering:
![image](https://user-images.githubusercontent.com/39925111/44037614-0b80a490-9f15-11e8-9ee5-ca83e8f1e375.png)

> wouldn't it be really tedious to deploy a butchering rack every time you find a shocker zombie?

Age old discussion of whats better: convenience and simplicity over depth and simulation. To eat a cookie and have a cookie. Since Kevin has the last word here, I followed his reasoning, read his comments above - he outlined the shape of this PR. Also have in mind that I have added two methods of using a rack, one of which is stationary but can be built with little effort, second is mobile and can be obtained here and there, and can be fabricated too. Same with table (or equivalent) - I even added a folding tourist table for convenience. I added kitchen unit as a equivalent of a table. None of which I was obliged to - but did anyway, knowing how many folks want more convenience. In this regard I believe I've done maximum that could be done, until/unless other ideas emerge.

Night-Pryanik commented:
Just in case: we can't perform full butchery after we did field dressing, right?

nexusmrsep commented:
You can perform full buthery after field dressing, even after quartering. Its the whole point here - field dressing prepares the corpse for later butchering. Also - field dressing does **not** ruin CBMs, so you can recover them in full butchery. Quartering however ruins CBM becouse you roughly cut the carcass into 4 pieces. Field dressing reduces weight of the corpse. Quartering "splits" weight and volume to 4 separate items (quartered corpses) - each will yield 1/4 of original yields. Game tracks each stage of butchering so for example full butchering will result in different yields with intact corpse, field dressed corpse and quartered corpse, etc.

Night-Pryanik commented:
> You can perform full buthery after field dressing, even after quartering.

Then that should require rethinking. Field dressing remove intestines such as stomach, but during full butchery we can still recover some intestines-related CBMs like `Expanded Digestive System` or `Internal Furnace` (which seems to be interconnected with one's digestive system) or `Metabolic Interchange`.

nexusmrsep commented:
Well, I can discuss with @kevingrenade the need/possibility to add a separate method/option to search for CBMs called "dissect" and move it there from full butchery. But there would be need for tools too. There is no reason why you would need to set of tools for full butchery just to get few more pieces of meat and not while removing delicate electronic devices woven inside the body without damaging them. I dont know if you read the code behind this PR (i always falsely assume that everyone do so) but fine cutting tools are required for full butcher to yield CBMs. Diamond coroneas - yeah, a single point CBM, but dont pick CBM that happen to be convenient example to make your point, becuse i can always counter with thermal dissipation cbm, and a ton others located all over the body.

nexusmrsep commented:
@Night-Pryanic I fully agree but I would wait for bionic slots to be merged to do it properly.

Xhuis commented:
I've been torn on this change for some time, and your staunch stance on it further drives me deeper down that path.

What is the gameplay reason for this change? Outside of solely realism, what purposes does it serve beyond depth for its own sake? An argument can even be made for filthy clothing, as it's still completely usable while dirty, and the feature can be disabled.

I struggle to understand what this adds to the game that the existing butchery system does not do well already. It feels like busywork. Now instead of butchering your corpse and taking what's left with you, you have to:

" Lug around a butcher rack and spend a long time performing a full butchery just to get the same results as before
* Do a quick butchery and receive tremendously fewer resources
* Field dress the corpse and have to carry the corpse *with you* 
* Quarter the corpse to lose most of its use just to make it portable 

As a long time player who plays more than 10 hours on an average week, I cannot see any enjoyment coming from this. I am by and large accepting of change, but as was the case with other such changes with their only real argument being "it's more realistic" then I would ask you make it a world option just like filthy clothing, and that you disable that option in Bright Nights because of how much it clashes with BN's intention.

sfsworms commented:
Presently (if you have a single body), you press B and butcher.

In the new system, as far as I can tell, to perform the equivalent actions (field butchery), you need to press B to open the butchery menu and then B again to field-butcher.

Zireael07 commented:
Oh ok, I get it. Maybe there might be a way to set your preferred kind of butchering to avoid having to go through the menu every time? (remember, butchering is used *A LOT* of times)

nexusmrsep commented:
Also field butchering reduces rot of extracted meat, rot ratio is significantly suppressed, so this is an incentive to field dress a deer, stick it somewhere and get fresh meat later and not go hunting every other day. I think I should suppress the rot ratio of field dressed corpse too, so a slight rework is to be expected.

As for CBM extraction i believe after this discussion that i will go towards a separate option after all. It will not clog the menu too much imho, and i see the point of doing so. 

Also I will try to provide some screenshots of the menu. I understand its hard to discuss something that is outlined by my description but cannot be yet seen in action (unless you compile yourelf an instance).

I also believe there will be need for yield tweaks and balancing afterwards, but I tried to get most of it adjusted. Harvest.json will be the first to go.

nexusmrsep commented:
Ok, some teaser screenshots of the menu:

![image](https://user-images.githubusercontent.com/39925111/44114647-1c233f90-a00c-11e8-8037-b19dde5e36bd.png)
![image](https://user-images.githubusercontent.com/39925111/44114714-4c896704-a00c-11e8-9632-c5d0c0e4d91b.png)
![image](https://user-images.githubusercontent.com/39925111/44114848-9a30cf9c-a00c-11e8-93cd-621a369e7a49.png)
![image](https://user-images.githubusercontent.com/39925111/44114977-f6d82cc2-a00c-11e8-968f-ba45db0ef359.png)
![image](https://user-images.githubusercontent.com/39925111/44115020-1252626a-a00d-11e8-92e6-1eb8f63a475f.png)

nexusmrsep commented:
Ok, lets sum up last changes:
1. **DISSECT** menu option for reconnaissance dissection of the corpse in search of CBM's, it is now the **ONLY** option for CBM harvesting
2. **dissecting** needs fine cutting tools, better tool = better chance
3. instead of survival **dissecting** depends and trains first aid (as a closest equivalent to anatomy), and knowledge in electronics can partially influence outcome (bonus)
4. **dissecting** destroys corpse
5. **dissecting** yields only CBMs - all other yields are lost (don't ask me to change that - I simply won't)
6. you get feedback on the quality of the tool you are using for this.
7. Quartered and badly field dressed corpses are too damaged for **dissecting**, but properly **field dressed** corpses can be successfully **dissected** (until bionic slots are merged I will not implement @Night-Pryanik  idea about yield/loss of CBMs that are located in guts or replace them )

**ALSO:**
Corpses that were properly **field dressed** will rot at 75% rate of normal rot. Harvested meat will rot at normal pace, but it's age will be reduced by same 75% factor, so starting rot will be reduced. Overall this means that butchered corpses will last longer and will yield fresh meat longer. This was planned & implemented from the beginning, I just revamped the code to do it for both corpse & meat alike.

Xhuis commented:
`until bionic slots are merged I will not implement`
`don't ask me to change that - I simply won't`

These comments, and others beforehand in the PR, emphasize the attitude that has me so iffy on this. You are very assured that this change is fully positive, and even after implementing a new feature, you immediately state that you will not change it. Why? As an open-source codebase, being open to change is part of the whole reason we operate the way we do.

Furthermore, even though I feel like I'm talking to a brick wall as you've rebuffed suggestion before it has even been made, why does dissection take so long? What does this add; why should it be the only source when some bionics are little more than batteries put into the body? There's a difficulty variable for a reason; why could simple, easy bionics not be recoverable from butchery and difficult ones not?

I'm not one to be confrontational, but this change has me concerned with if the game would be fun or not, and you seem staunchly against changing your course from the destination. My past experience in coding and following that same path tells me that you aren't making the right choice.

tyrael93 commented:
Oh hey you implemented the dissection thing. That's nice.
I'll be honest I wasn't really following through the whole thing here the only thing that irked me was needing a twble and a rack for getting bionics and that happened so cool beans

cainiaowu commented:
Well, feature got implemented first, and balance happened in latter PR, when someone is interested (or get annoyed enough) show up, that was how it had been for years (at least that is what I observed).

Night-Pryanik commented:
> when some bionics are little more than batteries put into the body? There's a difficulty variable for a reason; why could simple, easy bionics not be recoverable from butchery and difficult ones not?

This PR is already big enough. Ideally it should be divided into smaller parts (I can suggest some ways if @nexusmrsep would be interested). nexusmrsep (or you, or someone else) could add the feature you are talking about in other, later PRs.

But aside of that I think we should playtest his changes first, prior to saying that it won't be fun, or it will be tedious and so forth.

DracoGriffin commented:
>But aside of that I think we should playtest his changes first, prior to saying that it won't be fun, or it will be tedious and so forth.

Agreed. Having already compiled and tested, I think it's best to move this one forward (it's in a pretty good place) and then adjust with fine-tuning.

nexusmrsep commented:
@Xhuis you don't want to see the fact, that after a dozen of suggestions I have made dozens changes to make it better, and this PR's present state has nothing in common with what it was at the beginning. If you don't like it - fine. But I believe that your negative attitude towards this PR makes you judge things out of context.

The only reason why I will not implement CBM removal in field dressing *at this stage* because I would be forced to hard code it, but when bionic slots are done it will be more natural to discern / filter bionics by their. There is nothing more in it, I even said the the idea itself is reasonable.

When i said "dissecting yields only CBMs - all other yields are lost (don't ask me to change that - I simply won't)" - what i meant is that DISSECT option is a result of two alternative ideas - one is a separate option (dissect, and the other is making it a part of FULL BUTHERY. There is no third way for this, and since there were many requests that making it a part of FULL BUTCHERY is not a good thing, I had been convinced to change that and introduce DISSECT option. Since this might likely trigger comments and rant like "what? how come I can't get CBMs while full butchering? that's counter intuitive! change it now!" I simply stated that after listening to comments I made a **choice** and I will not change it just because someone might want it the other way around. And as long as I call the shots for my PR this is my right to do it.

Why dissection takes so much time to do? It's kinds obvious - player does not know what CBMs are implanted in the body. Perhaps some are visible, but those more discrete will not be obvious. Player is not a medical god so I assumed that he does not have has knowledge about every existing CBM, but wants to check the body limb by limb. This must be done in a careful way - every organ can have CBMs, even the skin itself. And after finding one - do you know how long invasive surgeries take? Answer: long time. Why? Because they are complicated. You don't butcher and rip out, yank, pluck out or tear out CBMs just like that. They are delicate electronic devices **surgically** implanted - they require surgical precision and care to remove them without damaging them, and this - without autodoc, in a field operation - has to consume time.

As for fine tuning - as @DracoGriffin and @Night-Pryanik suggest - this needs more testing. In that department I'm very reluctant to touch any of the values before it gets enough testing. I have not changed the way how the code calculate success (although this part is iffy to me), just changed skill used in 'dissect'. For example think that broken cyborg yields are too low, but maybe it was meant this way, so I don't wat to rush changes here. I need to learn how to perform automated tests in CATCH2 to get some statistics, put them under discussion and then fine tune the results to accepted values.

Xhuis commented:
I do agree that changes have been placed into the PR over the course of its lifespan, but I do not like the core changes. Before the addition of portable racks and tables, acquiring yields was a chore, and their existence serves as a bandage for the problem at large; even with these additions, the tourist table and butchering rack take up 25 L and 10 L respectively, making them impractical for transportation and making them serviceable mostly for carrying in a car.

Even without my own opinions, it suffers from the thing that the bionics changes got around: it is one massive, game-changing addition that is both difficult to review and difficult to assess as a bloated whole. Splitting it into parts is now impractical because all of the code has already been made.

Perhaps I just don't see something everyone else does, or I'm not of the same mindset as the other 14 people who have commented in this PR. Perhaps I'm not seeing the "great" part of it, the thing that makes it a cool change that seems to have so much support from everyone else. But I still struggle to understand what makes this strange, huge addition fun, and an interesting addition that the average player would find enjoyable instead of needlessly complex.

Too-DAMN-Much commented:
there needs to be some way to butcher identically to the old system, or at least not to have to sacrifice things in comparison to it.

truthfully quartering is useless and wasteful, that will never happen in my saves, same with field butchering, basically this is an astronomical carrying weight and volume nerf cleverly disguised, minus that though? sure i love it.

sfsworms commented:
Ypu haven't been paying attention. Field butchering is basically the old system (except you don't get offals and skin). Quartering can be useful if you don't have a vehicle and want to haul your kill to you base.

nexusmrsep commented:
@sfsworms Field butchering actually yields both offal and stomach. Its the removal of internal organs, so it would be strange if it wouldn't.

Also, the only downside of quartering at this point is loss of CBMs but you wouldn't even consider quartering if you aim. Worth mentioning - If you perform full butchery of all 4 parts you get from quartering you get full yield of the field dressed corpse (perhaps minus rounding down when dividing by 4).

Too-DAMN-Much commented:
@sfsworms no, i have, i just care about skin and offal.

nexusmrsep commented:
> field dressing loses the skin

No it doesn't. Quartering - yes it does. Do you have to quarter? No. Your choice.

>  i don't see it being realistic anymore

Quite the opposite.

> but being real here, CBM thing alone is enough to hate this.

@Too-DAMN-Much Just keep your hate outside this place. Please. You are not being real. You haven't done your homework and did not follow the changes, didn't read the code. You do not add anything of value to the discussion. You just want to keep status quo for the sake of your convenience. Sorry, but it is obvious you came here to troll. This in not a place for that. Go on Discord or Reddit, because this is GitHub, and it's not welcomed here.

Too-DAMN-Much commented:
updates have been ruining the game for a long time now.

fuck you all, don't care anymore if this is the way you're going to be.

just know that these changes ruined the game forever.

kevingranade commented:
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/dissecting-in-0-e/24327/18
